os/fs/driver/mtd/smart.c

```diff
@@ -5453,48 +5451,49 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 
 	fvdbg("address : %u\n", address);
 	type = GET_JOURNAL_TYPE(log->status);
-	/* TODO Recovery Logic Will be added, after recovery we should update dev structure too(cf.freecount) */
+	
 	switch (type) {
-	case SMART_JOURNAL_TYPE_COMMIT:
 		
-		break;
+	case SMART_JOURNAL_TYPE_COMMIT:
+		/* Journal is valid, so we will verify written data on data sector with crc of MTD header in journal */
+		memcpy(dev->rwbuffer, &log->mtd_header, mtd_size);
+		ret = MTD_READ(dev->mtd, psector * dev->sectorsize + mtd_size, dev->sectorsize - mtd_size, (FAR uint8_t *)&dev->rwbuffer[mtd_size]);
```

Comment:
how about this
```
int currReadSize = dev->sectorsize - mtd_size;
```
because it is used twice on lines 5460 and 5461.
It's just my opinion.

Reply:
If any code length is increased in specific line, I will remember your comment.
For now, I will keep current code. 
Thanks.